app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -36,7 +46,32 @@ def sso_callback # rubocop:disable Metrics/AbcSize
     alias mocksaml sso_callback
 
     def provider
-      request.env['omniauth.auth'].provider
+      request.env['omniauth.auth']&.provider || params[:provider]
+    end
+
+    def deauthorize # rubocop:disable Metrics/AbcSize
+      identity = current_user.identities.find_by(omniauth_provider: provider)
+      if is_navigational_format? && identity&.destroy
+        remove_provider_from_session(provider)
```

Comment:
I was thinking about the same before. When implementing the current solution, I tried avoiding much overhead, so decided that this would be fine: The wording is only about unlinking the account (which still exists), so a user who unlinked their last authentication method would need to request a new password via mail. Nonetheless, I fixed it altogether with the previous remark.